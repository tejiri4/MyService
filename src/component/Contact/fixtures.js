export const inputTypes = [
  {
    type: 'text',
    placeholder: 'Full name',
    key: 'fullname'
  },
  {
    type: 'email',
    placeholder: 'Email address',
    key: 'email'
  },
  {
    type: 'text',
    placeholder: 'Subject',
    key: 'subject'
  },
]

export const contactDetails = [
  {
    icon: 'https://res.cloudinary.com/store-manager/image/upload/v1563274461/facebook-placeholder-for-locate-places-on-maps.svg',
    alt: 'location',
    header: 'Location',
    body: '234 Ikorodu Road, Lagos, Nigeria.',
    showBorder: true
  },
  {
    icon: 'https://res.cloudinary.com/store-manager/image/upload/v1563274461/24-hour-support.svg',
    alt: 'phone-number',
    header: 'Phone number',
    body: '08114320486',
    showBorder: true
  },
  {
    icon: 'https://res.cloudinary.com/store-manager/image/upload/v1563274461/mail-black-envelope-symbol.svg',
    alt: 'email',
    header: 'Email address',
    body: 'tejirimatthew4@gmail.com'
  },
];