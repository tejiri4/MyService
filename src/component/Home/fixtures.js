export const socialMediaInfos = [
  {
    key: 'twitter',
    src: 'https://res.cloudinary.com/store-manager/image/upload/v1560940981/twitter.svg',
    alt: 'twitter-icon'
  },
  {
    key: 'instagram',
    src: 'https://res.cloudinary.com/store-manager/image/upload/v1560941286/instalgram.svg',
    alt: 'instalgram-icon'
  },
  {
    key: 'facebook',
    src: 'https://res.cloudinary.com/store-manager/image/upload/v1560941284/facebook.svg',
    alt: 'facebook-icon'
  }
]
