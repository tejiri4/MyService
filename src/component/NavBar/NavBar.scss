.nav-bar-component {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin-right: 50px;

  .active {
    color: #4D62D3;
  }

  nav {
    ul {
      display: flex;
      margin-right: 20px;
      list-style-type: none;
    }

    span {
      text-decoration: none;
      color: black;
      font-size: 11px;
      font-weight: 600;
      cursor: pointer;
    }

    li {
      margin-right: 20px;
    }
  }
}
