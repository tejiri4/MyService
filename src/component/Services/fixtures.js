export const ServiceCardDetails = [
  {
    img: 'https://res.cloudinary.com/store-manager/image/upload/v1561031570/photo-camera.svg',
    title: 'Photography',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
  },
  {
    img: 'https://res.cloudinary.com/store-manager/image/upload/v1561031570/website-design-symbol.svg',
    title: 'Web design',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eget iaculis augue.'
  },
  {
    img: 'https://res.cloudinary.com/store-manager/image/upload/v1561031571/programming.svg',
    title: 'Web application',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eget iaculis augue.'
  },
  {
    img: 'https://res.cloudinary.com/store-manager/image/upload/v1561031546/mobile.svg',
    title: 'Mobile application',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
  }
]
