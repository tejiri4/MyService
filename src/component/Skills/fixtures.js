export const skillsArray = [
  {
    skill: 'Photography',
    percentage: '90%',
    color: '#786DBB'
  },
  {
    skill: 'Web design',
    percentage: '70%',
    color: '#38DACF'
  },
  {
    skill: 'Web development',
    percentage: '90%',
    color: '#3088C7'
  },
  {
    skill: 'Mobile application',
    percentage: '80%',
    color: '#DE3AAF'
  }
];
