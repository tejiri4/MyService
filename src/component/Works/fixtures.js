export const works = [
  {
    name: "ALL",
    key: "all",
    data: [
      'https://res.cloudinary.com/store-manager/image/upload/v1562948655/balazs-ketyi-6ba_vdgx_go-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562948760/tracy-adams-rA5MU4pXnWE-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562948603/balazs-ketyi-LPWl2pEVGKc-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562948655/balazs-ketyi-6ba_vdgx_go-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562948760/tracy-adams-rA5MU4pXnWE-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562949962/ben-kolde-Ajcipx1VDXI-unsplash.jpg',
    ]
  },
  {
    name: "WEBDESIGN",
    key: "webdesign",
    data: [
      'https://res.cloudinary.com/store-manager/image/upload/v1562948655/balazs-ketyi-6ba_vdgx_go-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562948760/tracy-adams-rA5MU4pXnWE-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562948603/balazs-ketyi-LPWl2pEVGKc-unsplash.jpg',
    ]
  },
  {
    name: "WORDPRESS",
    key: "wordpress",
    data: [
      'https://res.cloudinary.com/store-manager/image/upload/v1562948655/balazs-ketyi-6ba_vdgx_go-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562948760/tracy-adams-rA5MU4pXnWE-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562948603/balazs-ketyi-LPWl2pEVGKc-unsplash.jpg',
    ]
  },
  {
    name: "BRANDING",
    key: "branding",
    data: [
      'https://res.cloudinary.com/store-manager/image/upload/v1562948655/balazs-ketyi-6ba_vdgx_go-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562948760/tracy-adams-rA5MU4pXnWE-unsplash.jpg',
      'https://res.cloudinary.com/store-manager/image/upload/v1562948603/balazs-ketyi-LPWl2pEVGKc-unsplash.jpg',
    ]
  }
];
