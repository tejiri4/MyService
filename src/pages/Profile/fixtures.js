export const navArray = [
  {
    name: "HOME",
    key: "home"
  },
  {
    name: "ABOUT ME",
    key: "about-me"
  },
  {
    name: "SERVICE",
    key: "service"
  },
  {
    name: "PORTFOLIO",
    key: "portfolio"
  },
  {
    name: "BLOG",
    key: "blog"
  },
  {
    name: "CONTACT",
    key: "contact"
  }
];
