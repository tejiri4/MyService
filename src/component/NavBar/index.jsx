// react libraries
import React, { useState } from 'react';
import PropTypes from 'prop-types';

// component
import Button from '../Button/index';

// scss
import './NavBar.scss';

const NavBar = props => {
  const [activeNavBar, setActiveNavBar] = useState('home');

  const { navArray, showButton } = props;

  const handleNavLinkClick = key => () => setActiveNavBar(key);

  const navLink = navDetail => (
    <li key={navDetail.key}>
      <span onClick={handleNavLinkClick(navDetail.key)} className={`${navDetail.key === activeNavBar && 'active'}`}>
        {navDetail.name}
      </span>
    </li>
  );

  return (
    <div className='nav-bar-component'>
      <nav >
        <ul>
          {navArray.map(navDetail => {
            return navLink(navDetail);
          })}
        </ul>
        {
          showButton && <Button
            name='Hire'
            width='100px'
            height='40px'
            color='#1367fb'
            borderColor='#1367fb'
          />
        }
      </nav>
    </div>);
};

NavBar.propTypes = {
  showButton: PropTypes.bool,
  navArray: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      key: PropTypes.string
    }),
  ),
};

export default NavBar;
