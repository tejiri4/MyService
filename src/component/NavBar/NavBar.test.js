// react libraries
import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './';
import { navArray } from '../../pages/Profile/fixtures';

describe('NavBar component', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      navArray,
      showButton: true,
      selectedNavElement: () => {}
    };
    wrapper = shallow(<NavBar {...props} />);
  });

  it('renders the NavBar component without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleNavLinkClick event handler', () => {
    const navlink = wrapper.find('span').last();
    navlink.simulate('click');
    expect(wrapper.find('.active').length).toEqual(1);
  });
});

