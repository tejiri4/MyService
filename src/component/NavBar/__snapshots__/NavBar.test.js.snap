// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders the NavBar component without crashing 1`] = `ShallowWrapper {}`;
